### PR TITLE
Update implementation of app gateway

### PR DESCRIPTION
**Pull request by steves-bjss**
Update terraform to add App Gateway

### CODE REVIEW

**Review by MichaelRobinson-BJSS — DISMISSED** [review]

File: terraform/app_gateway_ingress.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  dns_name                = "gpitfbuyingcatalogue${local.shortenv}"
+  pip_name                = "${var.project}-${var.environment}-publicip"
+  app_service_hostname    = module.webapp.webapp_default_site_hostname
+  
+  depends_on = [
+    azurerm_user_assigned_identity.managed_id
+    # azurerm_key_vault_access_policy.keyvault_aad_access,
+    # azurerm_network_security_rule.Azure,
+  ]
+}
```
Comment (PhysicsJon) [inline terraform/app_gateway_ingress.tf:24]: EoF newline missing

File: terraform/modules/bc_app_gateway_ingress/variables.tf [hunk 2]
```diff
@@ -37,3 +37,7 @@ variable "pip_name" {
 variable "managed_id_principal_id" {
   type = string
 }
+
+variable "app_service_hostname" {
+  type = string
+}
```
Comment (PhysicsJon) [inline terraform/modules/bc_app_gateway_ingress/variables.tf:43]: EoF newline missing

File: terraform/modules/bc_webapp_pri/output.tf [hunk 3]
```diff
@@ -21,3 +21,8 @@ output "webapp_default_site_hostname" {
     value = azurerm_app_service.webapp.default_site_hostname
     sensitive   = false
 }
+
+output "webapp_service_id" {
+    description = "The Web App service ID"
+    value = azurerm_app_service.webapp.id
+}
```
Comment (PhysicsJon) [inline terraform/modules/bc_webapp_pri/output.tf:28]: EoF newline missing

File: terraform/modules/bc_webapp_pri/variables.tf [hunk 4]
```diff
@@ -98,4 +98,8 @@ variable "smtp_server_host" {
 
 variable "smtp_server_port" {
   type = string
+}
+
+variable "vnet_subnet_id" {
+  type = string
 }
```
Comment (PhysicsJon) [inline terraform/modules/bc_webapp_pri/variables.tf:105]: EoF newline missing

File: terraform/network_security_group_association.tf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+resource "azurerm_subnet_network_security_group_association" "gateway" {
+  subnet_id                 = azurerm_subnet.gateway.id
+  network_security_group_id = azurerm_network_security_group.gateway.id
+
+  depends_on = [
+    azurerm_network_security_rule.Azure
+  ]
+ }
+
```
Comment (PhysicsJon) [inline terraform/network_security_group_association.tf:9]: EoF newline missing

File: terraform/virtual_network_subnet.tf [hunk 6]
```diff
@@ -2,7 +2,23 @@ resource "azurerm_subnet" "gateway" {
...
+      name = "${var.project}-${var.environment}-backend-delegation"
+
+      service_delegation {
+        name    = "Microsoft.Web/serverFarms"
+        actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+      }
+  }
 
   service_endpoints    = ["Microsoft.Sql","Microsoft.Storage"]
 }
```
Comment (PhysicsJon) [inline terraform/virtual_network_subnet.tf:24]: EoF newline missing

File: terraform/webapp_virtual_network.tf [hunk 7]
```diff
@@ -0,0 +1,4 @@
+resource "azurerm_app_service_virtual_network_swift_connection" "webapp" {
+  app_service_id = module.webapp.webapp_service_id
+  subnet_id      = azurerm_subnet.backend.id
+}
```
Comment (PhysicsJon) [inline terraform/webapp_virtual_network.tf:4]: EoF newline missing

**Review by PhysicsJon — CHANGES_REQUESTED** [review]
Looks sensible enough, just got a few missing EoF newlines

File: terraform/webapp_virtual_network.tf (hunk 7 above)
Comment (steves-bjss) [reply terraform/webapp_virtual_network.tf]: Updated

File: terraform/network_security_group_association.tf (hunk 5 above)
Comment (steves-bjss) [reply terraform/network_security_group_association.tf]: Updated

File: terraform/virtual_network_subnet.tf (hunk 6 above)
Comment (steves-bjss) [reply terraform/virtual_network_subnet.tf]: Updated

File: terraform/modules/bc_webapp_pri/variables.tf (hunk 4 above)
Comment (steves-bjss) [reply terraform/modules/bc_webapp_pri/variables.tf]: Updated

File: terraform/modules/bc_webapp_pri/output.tf (hunk 3 above)
Comment (steves-bjss) [reply terraform/modules/bc_webapp_pri/output.tf]: Updated

File: terraform/modules/bc_app_gateway_ingress/variables.tf (hunk 2 above)
Comment (steves-bjss) [reply terraform/modules/bc_app_gateway_ingress/variables.tf]: Updated

File: terraform/app_gateway_ingress.tf (hunk 1 above)
Comment (steves-bjss) [reply terraform/app_gateway_ingress.tf]: Updated

**Review by PhysicsJon — APPROVED** [review]

**Pull request merged**